File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/ItemCatService.java
package com.pinyougou.sellergoods.service;

import java.util.List;

import com.pinyougou.pojo.TbItemCat;
/**
 * 业务逻辑接口层
 * @author 89568
 *
 */
public interface ItemCatService {
	/**
	 * 分页
	 * @param parenid
	 * @return
	 */
	public List<TbItemCat> getItemCatByparenId(Long parenid);
	/**
	 * 新增和修改
	 * @param itemCat
	 */
	public void addOrupdata(TbItemCat itemCat);
	/**
	 * 根据主键id查找
	 * @param id
	 * @return
	 */
	public TbItemCat findOne(Long id);
	/**
	 * 删除
	 * @param id
	 */
	public void deleteItemCatById(Long[] id);
	/**
	 * 查找全部
	 */
	public List<TbItemCat> findAll();
}

File: pinyougou-manage-web/src/main/webapp/js/service/ItemCatService.js

app.service("ItemCatService",function($http){
	//根据父类id获取商品分类信息
	this.getItemCatByparenId = function(parenId){
		return $http.get("../ItemCat/select.do?id="+parenId);
	}
	
	//新增或者修改
	
	this.addOrupdata = function(entity){
		return $http.post("../ItemCat/addOrupdata.do",entity);
	}
	
	//根据主键id查找
	this.findOne = function(id){
		return $http.get("../ItemCat/findOne.do?id="+id);
	}
	
	//根据主键id删除
	this.delectByItemCatId = function(id){
		return $http.get("../ItemCat/deleteById.do?id="+id)
	}
	
	this.findAll = function(){
		return $http.get("../ItemCat/selectAll.do");
	}
})